lib/registries/addon/components/registries-banner/component.ts

```diff
@@ -12,29 +12,28 @@ import pathJoin from 'ember-osf-web/utils/path-join';
 import styles from './styles';
 import template from './template';
 
-const stateToBannerMap: Record<RegistrationState, {text: string, type: string}> = {
-    PendingRegistration: {
+const banner: Record<string, {text: string, type: string}> = {
+    pendingRegistration: {
```

Comment:
I'm confused -- why not switch to `[RegistrationState.PendingRegistration]: { ... }`? This component is necessarily coupled with the set of possible states; it seems clearer to express that coupling here than in a repetitive, 30-line computed property.

Reply:
Well less lines !== better code.  But these concepts are coupled but I would argue unnecessarily. The first issue with it is, you have to make sure the properties of the object match the enum (which we shouldn't do). The second issue is that you have to create redundant states inside of this "stateToBannerMap" - look at the withdraw and public states. They're both empty but yet we have to keep the concepts similar (even though we don't need to) because they're coupled together. This is basically just showing UI elements based on some states but we should couple our UI to some model state logic. They can easily be decoupled, we can extend the UI way more easily (think about if we change the UI but not the states - we'd have to zero out an object), and we won't break things by changing one thing in another place but having weird knockon effects in what is a logically unconnected place.

Reply:
I can't really see a good argument for coupling your UI logic with your state logic and treating them similar.

Reply:
Ok, fair points. Was thinking more from the perspective of "easy to read and understand how things are connected" than "easy to change something without worrying about its connections". I can see how this is better.

Sidenote: Given that, I'd remove the type annotation on `banner` -- literal objects have literal types, which provides some safety against typos/changes. Not all that important here, because the banner component won't be around much longer, but just fyi.

Reply:
Sorry, what do you mean by type annotation? Also, we could probably just return the objects with out the record at the top.

Reply:
I mean instead of
```ts
const banner: Record<string, {text: string, type: string}> = { ...
```
just
```ts
const banner = { ...
```
TS will use a literal type for the object, so you'd get a compile-time error for typos like:
```ts
const { perdingRegistration } = banner;
```
With `Record<string, ...>` it assumes any string key is fine.

But yeah, with how you've done the decoupling, no real need for the separate map.

Reply:
gotcha